analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/ValidatorTestCheck.cs

```diff
@@ -102,6 +109,10 @@ protected override ProgramState PostProcessSimple(SymbolicContext context)
                 (invocation.TargetMethod.IsStatic || invocation.Arguments.Length == 1).Should().BeTrue("Tag method should be static to not infer with object states.");
                 AddTagName(invocation.Arguments.First().Value.ConstantValue, context);
             }
+            if (context.Operation.Instance.TrackedSymbol() is { } symbol)
+            {
+                symbols[symbol.Name] = symbol;
```

Comment:
The symbol name can be ambiguous (e.g. shadowed fields). You may want to guard against that (e.g. an assertion, that any symbol of the same name already in the dictionary is the same as `symbol`).

Reply:
No, I intentionally allowed that so we can cherry pick anything that is used last.